iroh-qrpc-types/src/addr.rs

```diff
@@ -0,0 +1,117 @@
+use anyhow::{anyhow, bail};
+use quic_rpc::{transport::mem, Service};
+use serde_with::{DeserializeFromStr, SerializeDisplay};
+use std::{
+    fmt::{Debug, Display},
+    net::SocketAddr,
+    str::FromStr,
+};
+
+/// An address. This can be either a memory address, already containing the channel, or a network
+/// address which will have to be opened.
+#[derive(SerializeDisplay, DeserializeFromStr)]
+pub enum Addr<S: Service> {
+    Http2(SocketAddr),
+    Http2Lookup(String),
+    Mem(
+        mem::ServerChannel<S::Req, S::Res>,
+        mem::ClientChannel<S::Res, S::Req>,
+    ),
```

Comment:
note entirely sure how much i like that these are always together.  it means you can't guarantee that someone will only use the client-side i think?

Reply:
The way it was used before, there was a way to flip from client to server and back, which just did not work for mem channels. At least this way it is consistent.

As long as somebody has access to the crate you can not stop him from creating a server or client channel given an address anyway...